### PR TITLE
Graphql support

### PR DESCRIPTION
**Pull request by jhomarolo**
Add support to GraphQL using Apollo Server

### CODE REVIEW

File: src/api/graphql/resolvers/todoList.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const { getLists } = require('../../../usecases/getLists');
+const { createList } = require('../../../usecases/createList');
+
+function aUser({ hasAccess }) {
+  return { canCreateList: hasAccess };
+}
+
+const resolvers = {
+  Query: {
+    lists: async (parent, args) => {
```
Comment (dalssoft) [inline src/api/graphql/resolvers/todoList.js]: I think we should break each resolver on its own file

File: src/api/graphql/schema/todoList.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const { gql } = require('apollo-server-express');
+
+module.exports = gql`
+  extend type Query {
+    lists(ids: [Int]): [TodoList]
+  }
+
+  extend type Mutation {
+  
+  createList(
```
Comment (dalssoft) [inline src/api/graphql/schema/todoList.js]: a file for each type / mutation?

File: src/api/settings.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const dotenv = require('dotenv');
+dotenv.config({ silent: true });
+
+const Settings = {
```
Comment (dalssoft) [inline src/api/settings.js]: should there be a file for each env?

File: src/usecases/createList.js [hunk 4]
```diff
@@ -1,29 +1,32 @@
-const { Ok, Err, usecase, step, ifElse } = require('buchu')
-const { TodoList } = require('../entities/todoList')
+const { Ok, Err, usecase, step, ifElse } = require('buchu');
```
Comment (dalssoft) [inline src/usecases/createList.js]: modern JS usually don't use `;`

Comment (dalssoft) [reply src/usecases/createList.js]: is it possible to reinforce it using lint?

**Comment by jhomarolo** [comment]
Change folder architecture to separate files graphql

**Comment by jhomarolo-vortx** [comment]
I added some changes to your comments regarding the pull request and committed

**Comment by jhomarolo** [comment]
I added some changes to your comments regarding the pull request and committed

**Pull request merged**